Developer

Helli1995 opened the pull request:
Implemented some of the suggestions in rtpghi~.c to get more familiar with git discipline.

Helli1995 committed:
Update

Change location of rtpghi_init function-call to pd-objects DSP-method.

Helli1995 committed:
Update

Change signal_outlets to DSP-add function

Helli1995 committed:
Remove W user input

Remove W (num of audiochannels) as user input variable and uncomment memory allocation for complex output coefficients

Helli1995 committed:
Change length of c[]

According to phaseret src code function -rtpghi_execute(), the complex output array's size is size of M2, which is M/2+1

implement zero padding in rtpghi_perform in while loop

Helli1995 committed:
bug fix counter

correct counter in perform function (if_statment)

Helli1995 committed:
bug fixes

small compile error bug fixes

Helli1995 committed:
cast input sig s to t_sample* not float*

Helli1995 committed:
corrected argv in pdnew

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,80 +12,65 @@
+		if (e == 0) {
+			while (n--) {
+				
+				if (n >= (x->M_pd)/2-1){
```
i'm not sure i see the point of this.
does `phaseret_rtpghi_execute_s` return the full spectrum (including the (anti)symmetric 2nd half)? if so, copy all the bins (`[rifft~]` will discard the 2nd half anyhow)

Helli1995 replied:
Okay, now it seems to me that both, s[] and c[] should have length of M/2 +1, because they are using always the ltfat_int M2 = M/2 +1 (in rtpghi_execute, and inside the last function rtpghimagphase(), which is combining magnitude and phase to complex array). Also in the rtpghi_offline it is visible, or am I wrong with this? [https://github.com/ltfat/phaseret/blob/master/libltfat/modules/libphaseret/src/rtpghi.c]

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,80 +12,65 @@
+			}
+		}
+		else {
+			post("error of type: %d", e);
```
`pd_error(x, ...)`

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
     else {
         post("failed to init, status %d", init_s);
```
`pd_error()`

umlaeute commented on src/rtpghi~.c:
```diff
@@ -110,8 +91,9 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
-    init_s = phaseret_rtpghi_init_s(gamma, W, a , M, tol, do_causal, &(x->sta_pd));
+	ltfat_int w = 1;
+	
+    init_s = phaseret_rtpghi_init_s(gamma, w, a , M, tol, do_causal, &(x->sta_pd));
```
you might want to call `phaseret_rtpghi_init_s` only if the parameters have changed, so that when you turn the DSP off and then on, the state is preserved (the DSP might be restarted for whatever reasons and i guess that re-initializing the phaseret_state will cause a click)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -110,8 +91,9 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
-    init_s = phaseret_rtpghi_init_s(gamma, W, a , M, tol, do_causal, &(x->sta_pd));
+	ltfat_int w = 1;
+	
+    init_s = phaseret_rtpghi_init_s(gamma, w, a , M, tol, do_causal, &(x->sta_pd));
```
also, `phaseret_rtpghi_init` allocates a new phaseret_state, so you have any previously allocated state.

umlaeute replied:
please resolve these issues with (pushed) commits, and not by simply marking them as "resolved" (alternatively, add a comment why you think it ought not be resolved)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+	phaseret_rtpghi_done_s(&(x->sta_pd));
+
+	freebytes(x->c, (M/2+1) * sizeof *(x->c));
+	freebytes(x->out_real, sizeof (t_sample*));
```
`x->out_real` (and `x->out_imag`) are of type `t_outlet*`, not of type `t_sample*`!

anyhow, the two members are not used anywhere (except for allocating/deallocating the wrong type of memory), and should be removed.

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+  t_rtpghi_tilde *x = (t_rtpghi_tilde *)pd_new(rtpghi_tilde_class);
+
+    x->ol_pd=atom_getfloat(argv);
+    x->M_pd=atom_getfloat(argv+1);
```
you must check the number of arguments before accessing them.

something like

```C
switch (argc) {
  case 5: default:
     x->window_type=atom_getsymbol(argv+4);
  /* fallthrough */
  case 4:
    x->do_causal_pd=atom_getfloat(argv+3);
  /* fallthrough */
  case 3:
  // ...
}
```

alternatively you could use `atom_getfloatarg()`, but then you cannot assign default values != 0.+

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+    x->M_pd=atom_getfloat(argv+1);
+    x->tol_pd=atom_getfloat(argv+2);
+    x->do_causal_pd=atom_getfloat(argv+3);
+    x->window_type= *atom_getsymbol(argv+4);
```
that should be `x->window_type=atom_getsymbol(argv+4);` (see my comment on the `window_type` member.

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,80 +12,65 @@
+	t_float M_pd;
+	t_float tol_pd;
+	t_float do_causal_pd;
+	t_symbol window_type;
```
that should be a point to a symbol (`t_symbol*window_type;`)

also i'm not exactly sure about the naming scheme (why is there no `_pd` suffix?)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,80 +12,65 @@
+
+void rtpghi_tilde_dsp(t_rtpghi_tilde *x, t_signal **sp)
+{
     ltfat_int M = x->M_pd;
```
shouldn't `M` should be taken from the signal vector size?

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
-    post("x->M_pd: %d", M);
-    freebytes(x->c, (M) * sizeof *(x->c));
-    freebytes(x->out,2 * sizeof(t_sample *));
+	ltfat_int M = x->M_pd;
```
??

umlaeute commented on src/rtpghi~.c:
```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+                              0);
+   class_addmethod(rtpghi_tilde_class,
+                   (t_method)rtpghi_tilde_dsp, gensym("dsp"), A_CANT, 0);
+   CLASS_MAINSIGNALIN(rtpghi_tilde_class, t_rtpghi_tilde, f);
```
also add `class_addmethod(rtpghi_tilde_class, (t_method)rtpghi_tilde_causal, gensym("causal"), A_FLOAT, 0);` and friends.

umlaeute reviewed (commented):
see my notes

umlaeute commented on .DS_Store:
do not commit `.DS_Store` files.

people should be free to arrange the files in their finder as they like it.

umlaeute replied:
fix with https://github.com/Helli1995/PD-RTPGHI/pull/3/commits/270f8ee7ea4d8de6ba81ac1e1d2c17c8f3d526a4

Helli1995 committed:
Update post("")

change post command to pd_error, where an error should be thrown

Helli1995 committed:
bug fix memory init

destroy rtpghi state before initializing a new one, when DSP is tuned OFF and ON

Helli1995 committed:
Removal of members

removal of class_members out_real, out_imag from data_space (and init and free method), reason: unused

Helli1995 committed:
Update

Changed type and name of variable window_type

Helli1995 committed:
removing M_pd from class

remove blocksize user input varibale M_pd from object, should be chosen automatically.

leaving overlap (ol_pd) right now for debugging purposes, not shure how it interacts with external function rtpghi_execute

Helli1995 committed:
pd_new

added switch statement to pd_new to allow variable num of input arguments, setting default values

Helli1995 committed:
Delete .DS_Store

Helli1995 committed:
update_free method

only free space in mem, if there has been a prior allocation

Helli1995 committed:
Update DPS_method

Changed location of post/pd_error_msg

Helli1995 committed:
Debug DSP freeze

adding post("..")  for debugging

Helli1995 committed:
Update *rtpghi_tilde_new

implementation of variable amnt of input args, check number of args

Helli1995 committed:
add print for debugging

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Create .gitignore

Helli1995 committed:
Merge branch 'local' into developer

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,147 +12,187 @@
+		post("destroyed state at adrr: %p\n", &(x->sta_pd));
+		phaseret_rtpghi_done_s(&(x->sta_pd));
+		init_s = phaseret_rtpghi_init_s(gamma, w, a , M, tol, do_causal, &(x->sta_pd));
+		if (init_s == 0) {
```
it is of course possible to do it like this, but it's not very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), as you have to call the init func twice and make sure that you do it correctly all the time.

what you are doing right now is:
```C
if(x->resource) {
  free_resource(x->resource);
   x->resource = allocate_resource();
   check_resource(x->resource);
} else {
   x->resource = allocate_resource();
   check_resource(x->resource);
}
```

whereas a simplified version (that does the very same thing without duplicating code) looks like this:

if(x->resource) {
  free_resource(x->resource);
}
x->resource = allocate_resource();
check_resource(x->resource);
```

in practice, this would look more like this:

```C
if(changed) {
   free_resource(x->resource);
   x->resource = NULL;
}
if(!x->resource) {
   x->resource = allocate_resource();
   check_resource(x->resource);
}
```

umlaeute commented on src/rtpghi~.c:
```diff
@@ -12,147 +12,187 @@
+	t_object  x_obj;
+	t_float ol_pd;
+	t_float tol_pd;
+	t_float do_causal_pd;
```
any specific reason why this boolean value is a `t_float`?

umlaeute reviewed (commented):
getting better :-)

Helli1995 committed:
simplify code of memalloc for rtpghi_state

change order of alloc and free

Helli1995 committed:
Update rtpghi_tilde_new()

Change execution order of default values and switch () statement to make code more clear and effective.
Remove win_def from data_space

Helli1995 committed:
fixed bug dsp add

changed arg number from 3 to 5

Helli1995 committed:
Debug

Removed some dbug prints

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Upload new test patch

it did not crash (alsways) with higher overlap (-> smaller hopsize a for phaseret lib objects), which does not really make sense to me at the moment

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Update DSP_perform

added array s2 (s/2+1) to function and use of rtpghi_execute()

Helli1995 committed:
Added some comments

Uncomment: Divide blocksize (sp[0]->s_n) by 8 to give enough calculation time for rtpghi_execute() execute, of course no solution for final code because of mismatch of fft-sizes.

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
fix bug input arg order

fix bug rtpghi_init(): change location of gamma from 1 to 4 due to mismatch of documentation and implementation of rtpghi_init()

Helli1995 committed:
changed length of c and s back to M

dsp not crashing anymore

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Add method for set_causal and reset

Add method for set_causal function and reset fuction,
what to do with tolerance? It is not a libphaseret function, so maybe not meant to be changed, while DSP turned ON.

Helli1995 committed:
Removing some unnecessary prints

Helli1995 committed:
Update

Change to correct function names

Helli1995 committed:
Update set_tol and set_causal

Updatet methods again due to outdated documentation in libphaseret

Helli1995 committed:
Remove s2 variable

not needed

umlaeute commented on src/rtpghi~.c:
```diff
@@ -193,8 +189,16 @@ void rtpghi_tilde_setup(void) {
+	//to do: maybe change tol_pd on runtime (while DSP turned ON)
+	class_addbang  (rtpghi_tilde_class, rtpghi_tilde_res);
+	class_addmethod(rtpghi_tilde_class,
+			(t_method)rtpghi_tilde_causal, gensym("set_c"),
```
i think `set_c` is really a bad name. e.g. `set_c 1` tells us, that we are **set**ing the value of **c** to **123**. but what is `c` in the first place? a *c*ounter?

otoh, having a message `causal 1` tells us to do *something* with **causal**ity and the value **1**. in the absence of any explicit specifier what this *something* is, i think it is a safe bet to assume that we are *setting* the *causal* value to *1* (which is exactly what is happening).

feel free to use cryptic variables in your implementation (i *might* still object :-)), but make sure that your public API is verbose and self-explanatory.

umlaeute commented on src/rtpghi~.c:
```diff
@@ -135,11 +118,26 @@ void rtpghi_tilde_dsp(t_rtpghi_tilde *x, t_signal **sp)
 
+void rtpghi_tilde_causal(t_rtpghi_tilde *x, t_floatarg f)
+{
+	if ((f == 0.) || (f == 1.)) {
```
in Pd-land the typical convention for booleans is `!=0.0` (cf `[spigot]`, `[metro]`,...), so that `0.1` and `2` and `-666` are all truish.

umlaeute commented on src/rtpghi~.c:
```diff
@@ -193,8 +189,16 @@ void rtpghi_tilde_setup(void) {
+			A_DEFFLOAT, 0);
+	/*
+	class_addmethod(rtpghi_tilde_class,
+			(t_method)rtpghi_tilde_tol, gensym("set_t"),
```
same as above.

it really makes patches easier to read if they spell `[tolerance 0.54, causal 1(` instead of `[set_t 0.54, set_c 1(`


apart from keeping yourself sane, this also follows best practice as laid out by other built in objects and extra objects (such as `sigmund~`).
i find 0 occurences of the `set_<shortname>` message pattern in the Pd core code and only *very* few occurences of the `set<longname>` pattern, presumably for good reasons.

umlaeute reviewed (commented):
nice.

i've added some comments about making the behaviour more Pd-idiomatic.

Helli1995 committed:
Update causal, tolerance method

change message name

Helli1995 committed:
Example of usage

Spectral filter with PD-extension GEM.

Helli1995 committed:
debug

Helli1995 committed:
remove old examples

Helli1995 committed:
added pd help file

creation and implementation of helpfile

Helli1995 committed:
implemnt macros for PD floatsize

32 or 64 bit

Helli1995 committed:
init state only once

initialize a new rtpghi state only if the DSP is started for the first time, destroy state only, if object is destroyed or pd is closed

Helli1995 committed:
init if blocksize changes

Helli1995 committed:
adding method for overlap factor change

Helli1995 committed:
added method for change of window

Helli1995 committed:
Update rtpghi~.c

Helli1995 committed:
correcting some words

umlaeute commented on src/rtpghi~.c:
```diff
@@ -1,14 +1,30 @@
+  /* normally, our floats (t_float, t_sample,...) are 32bit */
+# define PD_FLOATSIZE 32
+#endif
+
```
`PD_FLOATSIZE` has been defined in Pd-headers since 0.45-0 (released more than 10 years ago).
it's probably safe to assume that it is already defined (and just error out if it is missing).

otoh, there's no harm done.

umlaeute commented on src/rtpghi~.c:
```diff
@@ -1,14 +1,30 @@
+#define phaseret_rtpghi_set_tol phaseret_rtpghi_set_tol_s
+#define phaseret_rtpghi_set_causal phaseret_rtpghi_set_causal_s
+#define phaseret_rtpghi_execute phaseret_rtpghi_execute_s
+
```
yes, something like this.
afair, you probably also want to define `LTFAT_DOUBLE` vs `LTFAT_SINGLE`  *before* including `phaseret.h`.
and then just do:
```C
#define rtpghi_init PHASERET_NAME(rtpghi_init)
// ...
```

and you just se `rtpghi_init()` instead of `phaseret_rtpghi_init_s()`.

the advantage over your solution is that you only need to `#define` once and both floatsizes are handled automatically (less code-duplication)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -56,10 +73,13 @@ t_int *rtpghi_tilde_perform(t_int *w) {
+	//post("start DSP");
     ltfat_int M = (ltfat_int) sp[0]->s_n;
+	if ((x->blocksize) == 0) {
+		(x->blocksize) = M;
```
very minor nitpick, but what is it with your indentation?
is it just your editor that handles tabs and spaces separately (with a tab having the same indentation as 4 spaces), or is there some hidden meaning?

it's driving me crazy :-/

umlaeute commented on src/rtpghi~.c:
```diff
@@ -18,6 +34,7 @@ typedef struct _rtpghi_tilde {
 	t_symbol *window_type_pd;
 	phaseret_rtpghi_state_s* sta_pd;
 	ltfat_complex_s *c;
+	ltfat_int blocksize;
```
here we still have two single-precision members (which won't work with `PD_FLOATSIZE=64`).

umlaeute commented on src/rtpghi~.c:
```diff
@@ -131,16 +153,92 @@ void rtpghi_tilde_tol(t_rtpghi_tilde *x, t_floatarg f)
+	if ((f > 0.0) && (f <= x->blocksize)) {
+		phaseret_rtpghi_done(&(x->sta_pd));
+		post("destroyed state at adrr: %p\n", &(x->sta_pd));
+		x->sta_pd=NULL;
```
it's probably better to refactor the (conditional) recreation into a separate function, that checks whether (at least) one of the constituent parameters has changed and (re)creates the `phaseret_rtpghi_state*` object.

something like:
```
void rtpghi_tilde_recreatestate(t_rtpghi_tilde*x, LTFAT_FIRWIN win, int blocksize, int overlap) {
   if(x->sta_pd && (win != x->win || blocksize != x->blocksize || overlap != x->overlap)) {
        phaseret_rtpghi_done(&(x->sta_pd));
        x->sta_pd=NULL;
   }
   x->win = win;
   x->blocksize = blocksize;
   x->overlap = overlap;
   if(!x->sta_pd) {
      if(phaseret_rtpghi_init(x->win, x->overlap, x->blocksize, ...)) {
         pd_error(x, "failed to init state");
          x->sta_pd=NULL;
      }
   }
}

umlaeute commented on src/rtpghi~.c:
```diff
@@ -169,6 +267,7 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
 	post("win_def %p, do_causal %f, tol %f, ol %f", x->window_type_pd, x->do_causal_pd, x->tol_pd, x->ol_pd);
     x->sta_pd=NULL;
     x->c = NULL;
+	x->blocksize = 0;
```
typically there's no need to set the members to `0`, as `pd_new()` already does this (but of course, there's no harm in doing so)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -194,15 +293,25 @@ void rtpghi_tilde_setup(void) {
 	class_addmethod(rtpghi_tilde_class,
 				   (t_method)rtpghi_tilde_dsp, gensym("dsp"), A_CANT, 0);
 	CLASS_MAINSIGNALIN(rtpghi_tilde_class, t_rtpghi_tilde, f);
+	class_sethelpsymbol(rtpghi_tilde_class, gensym("rtpghi~"));
```
useless. avoid using `class_sethelpsymbol` as much as possible (it's really only useful for multiple objects sharing the same help-patch (which should be avoided), and objects with names that are not legal filenames (e.g. `///`))

umlaeute commented on rtpghi~-help.pd:
```diff
@@ -0,0 +1,257 @@
+#X obj 47 96 noise~;
+#X obj 340 95 noise~;
+#X obj 664 131 noise~;
+#X msg 366 30 set 2048 16 1;
```
thanks for the much improved help-patch.

why do you explicitly set the blocskize/overlap/upsampling with a loadbanged message, when you could just use `[block~ 2048 16]` in the relevant subpatches?

it would be cool to have a:
- one radiobutton to select the reconstruction algorithm (phaseret, fft, none)
- one radiobutton to select the input signal
  (rather than multiple toggles that allow you to listen to the phaseret/noise at the same time as sine/fft and sine/none)
- have a musical input signal (e.g. by allowing the user to select a soundfile) - sines and noise typically do not allow you to assess the quality of an FFT-algorithm
- have a master fader, so you don't blow the PA

Helli1995 committed:
remove sethelpsymbol

Helli1995 committed:
change macros

Helli1995 committed:
Update Macros

Helli1995 committed:
recreate state function

- does not work with no arguments at init

Helli1995 committed:
Added function recreate state

added function and typedef member LTFAT_FIRWIN window

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Update helpfile

Add Radio slider

Add Master Fader, VU Meter

Helli1995 committed:
Remove post commands

some unnecessary prints

Helli1995 committed:
Update rtpghi~-help.pd

Helli1995 committed:
Update rtpghi~-help.pd

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Removed windows

Removed some windows in the list, because they do not work, changed description of window

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
abstraction testing

umlaeute commented on src/rtpghi~.c:
```diff
@@ -1,158 +1,243 @@
+
+	if (x->window_type_pd != NULL) {
+		x->window = ltfat_str2firwin(x->window_type_pd->s_name);
+	}
```
hmm, no. this doesn't really work as *i* imagined it.
if you set the members (`x->window`, `x->blocksize`) before calling `rtpghi_tilde_recreatestate()`,
the latter will not reallocate properly (as it now checks whether `x->window != x->window` which is obviously always true)

umlaeute commented on src/rtpghi~.c:
```diff
@@ -1,158 +1,243 @@
+	}
+}
+
+void rtpghi_tilde_window(t_rtpghi_tilde *x, t_symbol s) {
```
`s` really is a pointer `t_symbol*` (which kind of explains why it is crashing for me)

Helli1995 committed:
changed t_symbol*

updated helpfile

Helli1995 committed:
Added int window_check

Not a pretty, but functional solution to keep it all in one function (recreatestate)

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
remove files

removed file due to creation. of new patch

Helli1995 committed:
Delete shader.pd

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
upload debug file Gem

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Add. shader abstractions

Add PD-Abstractions for serial rendering on GFX-Card

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Added Turn On for gemheads

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Update recreatestate usage

replaced x->blocksize with M

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Changing comment in helpfile

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 commented:
Problem with the destructor method (occuring e.g. when using rtpghi~-helpfile.pd): The destructor works fine, when deleting the instance of the [rtpghi~] object, after selecting the rtpghi-algorithm (which automatically starts the DSP).
Although, starting the algorithm the same way and closing the helpfile (or removing the [pd-rtpghi] subpatch) crashes PD.

umlaeute commented:
> Problem with the destructor method 

which problem? how to reproduce it?

umlaeute commented:
> which problem? 

ah i see:

https://github.com/Helli1995/PD-RTPGHI/blob/d11de93091c2fa1314a480866244f26fa1db42ec/src/rtpghi~.c#L216

the destructor callback has a single argument, which is a reference to the object.

so `sp` really points to some undefined place, and dereferencing it (and worse: using a value from that limbo to free memory) in https://github.com/Helli1995/PD-RTPGHI/blob/d11de93091c2fa1314a480866244f26fa1db42ec/src/rtpghi~.c#L222 will kill a kitten.

Helli1995 committed:
Fix bug in destructor method

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Audio Processing Shader GUI

Load up first version

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Adding sources

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Add mirror and modifications

correct gaussian_blur

Helli1995 committed:
Updates

Update edge_det shader and add logarithm magnitude to pd

Helli1995 committed:
Fixed errors in Gaussian blur with size etc

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
Update mirror fragment shader and warping

add new uniforms and remove errors

Helli1995 committed:
update serial rendering help

correct bug in shader 1

Helli1995 committed:
Added self made Piano test file

changed window to blackman for testing f(from windowing lib)

Helli1995 committed:
Update shader and blocksize

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
some bug fixes and interpolation

Helli1995 committed:
Push last stable version

Due to discovered unstable version with crackling audio

Helli1995 committed:
Updating serial rendering

-Correct normalization of fft output
-Upload example images

Helli1995 committed:
Updatet normalization

Add factor 255 for GL_RGBA32F (8 bit exponent field)

Helli1995 committed:
Removing 255 from normalization

1. The value does not make sense
2. apparently the values are normalized to 0...1

Helli1995 committed:
Merge branch 'developer' of https://github.com/Helli1995/PD-RTPGHI into developer

Helli1995 committed:
Update GOS.pd

Helli1995 committed:
Merge branch 'local' into developer

Helli1995 committed:
minor changes

Helli1995 committed:
minor changes

Helli1995 committed:
Adding RTISI-LA algorithm

Helli1995 committed:
trying to include max_iter method

not compiling yet

Helli1995 committed:
Updating print messages and makefile

methods working now

Helli1995 committed:
changed some details in ode

Helli1995 committed:
try adc

Helli1995 committed:
update makefile name

phase_retrieval

Helli1995 committed:
Merge pull request #6 from Helli1995/more_subpatches

More subpatches

Helli1995 committed:
Changing lib name in patch

Helli1995 committed:
Update README.md

Add description for Mac USers

Helli1995 committed:
Merge pull request #7 from Helli1995/more_subpatches

Update README.md

Helli1995 committed:
Update normalization

move window_calc to super patch

Helli1995 committed:
Merge branch 'local' into more_subpatches

Helli1995 committed:
remove wav files

Helli1995 committed:
Merge pull request #8 from Helli1995/more_subpatches

move to dev

Helli1995 committed:
Merge branch 'developer' of https://github.com/Helli1995/PD-RTPGHI into developer

Helli1995 committed:
remove file
